counterpartylib/lib/messages/issuance.py

```diff
@@ -186,29 +236,78 @@ def compose (db, source, transfer_destination, asset, quantity, divisible, descr
+            if len(assets) > 0:
+                # this is a reissuance
+                asset = assets[0]['asset_name']
+            else:
+                # this is a new issuance
+                #   generate a random numeric asset id which will map to this subasset
+                asset = util.generate_random_asset()
+
+    call_date, call_price, problems, fee, description, divisible, reissuance, reissued_asset_longname = validate(db, source, transfer_destination, asset, quantity, divisible, callable_, call_date, call_price, description, subasset_parent, subasset_longname, util.CURRENT_BLOCK_INDEX)
     if problems: raise exceptions.ComposeError(problems)
 
     asset_id = util.generate_asset_id(asset, util.CURRENT_BLOCK_INDEX)
-    data = struct.pack(config.TXTYPE_FORMAT, ID)
-    if len(description) <= 42:
-        curr_format = FORMAT_2 + '{}p'.format(len(description) + 1)
+    if subasset_longname is None or reissuance:
+        # Type 20 standard issuance FORMAT_2 >QQ??If
+        #   used for standard issuances and all reissuances
+        data = struct.pack(config.TXTYPE_FORMAT, ID)
+        if len(description) <= 42:
+            curr_format = FORMAT_2 + '{}p'.format(len(description) + 1)
+        else:
+            curr_format = FORMAT_2 + '{}s'.format(len(description))
+        data += struct.pack(curr_format, asset_id, quantity, 1 if divisible else 0, 1 if callable_ else 0,
+            call_date or 0, call_price or 0.0, description.encode('utf-8'))
     else:
-        curr_format = FORMAT_2 + '{}s'.format(len(description))
-    data += struct.pack(curr_format, asset_id, quantity, 1 if divisible else 0, 1 if callable_ else 0,
-        call_date or 0, call_price or 0.0, description.encode('utf-8'))
+        # Type 21 subasset issuance SUBASSET_FORMAT >QQ?B
+        #   Used only for initial subasset issuance
+        # compacts a subasset name to save space
+        compacted_subasset_longname = util.compact_subasset_longname(subasset_longname)
+        compacted_subasset_length = len(compacted_subasset_longname)
+        data = struct.pack(config.TXTYPE_FORMAT, SUBASSET_ID)
+        curr_format = SUBASSET_FORMAT + '{}s'.format(compacted_subasset_length) + '{}s'.format(len(description))
+        data += struct.pack(curr_format, asset_id, quantity, 1 if divisible else 0, compacted_subasset_length, compacted_subasset_longname, description.encode('utf-8'))
+
     if transfer_destination:
         destination_outputs = [(transfer_destination, None)]
     else:
         destination_outputs = []
     return (source, destination_outputs, data)
 
-def parse (db, tx, message):
+def parse (db, tx, message, message_type_id):
     issuance_parse_cursor = db.cursor()
 
     # Unpack message.
     try:
-        if (tx['block_index'] > 283271 or config.TESTNET) and len(message) >= LENGTH_2: # Protocol change.
+        subasset_longname = None
+        if message_type_id == SUBASSET_ID and util.enabled('subassets', block_index=tx['block_index']):
```

Comment:
when `SUBASSET_ID` and subassets isn't enabled yet we shouldn't let it run through the `elif` and `else` that follow